client/qiskit_serverless/utils/runtime_service_client.py

```diff
@@ -0,0 +1,120 @@
+# This code is a Qiskit project.
+#
+# (C) Copyright IBM 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+======================================================================
+Json utilities (:mod:`qiskit_serverless.utils.runtime_service_client`)
+======================================================================
+
+.. currentmodule:: qiskit_serverless.utils.runtime_service_client
+
+Qiskit Serverless runtime client wrapper
+========================================
+
+.. autosummary::
+    :toctree: ../stubs/
+
+    ServerlessRuntimeService
+"""
+import os
+import logging
+from typing import Callable, Dict, Sequence, Type, Union, Optional
+
+import requests
+from qiskit_ibm_runtime import QiskitRuntimeService
+from qiskit_ibm_runtime.runtime_job import RuntimeJob
+from qiskit_ibm_runtime.runtime_job_v2 import RuntimeJobV2
+from qiskit_ibm_runtime.runtime_options import RuntimeOptions
+from qiskit_ibm_runtime.utils.result_decoder import ResultDecoder
+
+from qiskit_serverless.core.constants import (
+    REQUESTS_TIMEOUT,
+    ENV_JOB_GATEWAY_TOKEN,
+    ENV_JOB_GATEWAY_HOST,
+    ENV_JOB_ID_GATEWAY,
+    ENV_GATEWAY_PROVIDER_VERSION,
+    GATEWAY_PROVIDER_VERSION_DEFAULT,
+)
+
+
+def associate_runtime_job_with_serverless_job(
+    runtime_job_id: str, session_id: Optional[str] = None
+) -> bool:
+    """Make a request to gateway to associate runtime job id with serverless job id.
+
+    Args:
+        runtime_job_id (str): job id for runtime primitive
+        session_id (str): session/batch id
+
+    Returns:
+        bool: if request was ok
+    """
+    version = os.environ.get(ENV_GATEWAY_PROVIDER_VERSION)
+    if version is None:
+        version = GATEWAY_PROVIDER_VERSION_DEFAULT
+
+    token = os.environ.get(ENV_JOB_GATEWAY_TOKEN)
+    if token is None:
+        logging.warning("Runtime job will not be associated with serverless job.")
+        return False
+
+    url = (
+        f"{os.environ.get(ENV_JOB_GATEWAY_HOST)}/"
+        f"api/{version}/jobs/{os.environ.get(ENV_JOB_ID_GATEWAY)}/add_runtimejob/"
+    )
+    response = requests.post(
```

Comment:
Are we not using `safe_json_request` here for any reason like [here](https://github.com/Qiskit/qiskit-serverless/blob/28b27e4c5f26544f5fc751fc6ff5f32f32432421/client/qiskit_serverless/core/job.py#L386)?

(Thinking in maybe not need to sanitize the output).

Reply:
in this function we do not fail and do not need to raise any exceptions that is why vanilla requests is fine. If something goes wrong we need to warn in logs and that is it.

Reply:
Oh, I see. Like a best effort. Got it!